conans/test/functional/toolchains/cmake/cmakedeps/test_cmakedeps_find_module_and_config.py

```diff
@@ -0,0 +1,94 @@
+import textwrap
+
+import pytest
+
+from conans.test.assets.cmake import gen_cmakelists
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.assets.sources import gen_function_cpp, gen_function_h
+from conans.test.utils.tools import TestClient
+
+
+@pytest.fixture(scope="module")
+def client():
+    t = TestClient()
+    cpp = gen_function_cpp(name="mydep")
+    h = gen_function_h(name="mydep")
+    cmake = gen_cmakelists(libname="mydep", libsources=["mydep.cpp"])
+    conanfile = textwrap.dedent("""
+        import os
+        from conans import ConanFile
+        from conan.tools.cmake import CMake
+
+        class Conan(ConanFile):
+            name = "mydep"
+            version = "1.0"
+            settings = "os", "arch", "compiler", "build_type"
+            exports_sources = "*.cpp", "*.h", "CMakeLists.txt"
+            generators = "CMakeToolchain"
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+                cmake.build()
+
+            def package(self):
+                self.copy("*.h", dst="include")
+                self.copy("*.lib", dst="lib", keep_path=False)
+                self.copy("*.dll", dst="bin", keep_path=False)
+                self.copy("*.dylib*", dst="lib", keep_path=False)
+                self.copy("*.so", dst="lib", keep_path=False)
+                self.copy("*.a", dst="lib", keep_path=False)
+
+            def package_info(self):
+
+                self.cpp_info.set_property("cmake_find_mode", "both")
+
+                self.cpp_info.set_property("cmake_file_name", "MyDep")
+                self.cpp_info.set_property("cmake_target_name", "MyDepTarget")
+
+                self.cpp_info.set_property("cmake_module_file_name", "mi_dependencia")
+                self.cpp_info.set_property("cmake_module_target_name", "mi_dependencia_target")
```

Comment:
What about modules without targets? some of the older only use variables... how can we provide those?

Reply:
The issue with these modules without targets is, Which variables should we declare? probably nothing to be automated. This might be another reason to enable a property to introduce custom cmake code that would be declared when consuming the module/config but we didn't agree yet about that.